src/App.tsx

```diff
@@ -1,61 +1,144 @@
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(id: number) {
+  return usersFromServer.find(usr => (id) === usr.id) || null;
+}
 
 export const App = () => {
+  const initTodos = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUser(todo.userId),
+  }));
+
+  const [todos, setTodos] = useState<ToDo[]>(initTodos);
+  const [todo, setTodo] = useState<ToDo>(
+    {
+      id: 0,
+      title: '',
+      completed: false,
+      user: null,
+    },
+  );
+
+  const [isFilled, setIsFilled] = useState(true);
+
+  const onAdd = (newTodo: ToDo) => {
+    setTodos(currentTodos => [
+      ...currentTodos,
+      {
+        ...newTodo,
+        id: Math.max(...(todosFromServer.map(toDo => toDo.id))) + 1,
+      },
+    ]);
+  };
+
+  const resetForm = () => {
+    setTodo(
+      {
+        id: 0,
+        title: '',
+        completed: false,
+        user: null,
+      },
+    );
+
+    setIsFilled(true);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTodo(
+      { ...todo, title: event.target.value },
+    );
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setTodo(
+      {
+        ...todo,
+        user: getUser(+event.target.value),
+      },
+    );
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todo.title || !todo.user?.id) {
+      setIsFilled(false);
+
+      return;
+    }
+
+    onAdd(todo);
+    resetForm();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">
+            Title:
+            <input
+              type="text"
+              value={todo.title}
+              data-cy="titleInput"
+              name="title"
+              placeholder="Enter a title"
+              onChange={handleTitleChange}
+            />
+            {(!todo.title.trim() && !isFilled)
+              && (<span className="error">Please enter a title</span>)}
+          </label>
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label htmlFor="select">
+            User:
+            <select
+              name="select"
+              data-cy="userSelect"
+              onChange={handleUserIdChange}
+              value={!todo.user ? 0 : todo.user.id}
+            >
+              <option value="0" disabled>Choose a user</option>
+              {usersFromServer.map(usr => (
```

Comment:
`usr` is not such descriptive name as `user`, please rename argument in function